modules/aws/dc/dc-startup.ps1.tmpl

```diff
@@ -8,9 +8,28 @@ $LOG_FILE = "C:\Teradici\provisioning.log"
 $DATA = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
 $DATA.Add("admin_password", "${admin_password}")
 
+function Decrypt-Credentials {
+    try {
+        $ByteAry = [System.Convert]::FromBase64String("${admin_password}")
+        $MemStream = New-Object System.IO.MemoryStream($ByteAry, 0, $ByteAry.Length)
+        $DecryptResp = Invoke-KMSDecrypt -CiphertextBlob $MemStream
+        $StreamRead = New-Object System.IO.StreamReader($DecryptResp.Plaintext)
+        $DATA."admin_password" = $StreamRead.ReadToEnd()
+    }
+    catch {
+        "Error decrypting credentials: $_"
+        return $false
+    }
+}
+
 Start-Transcript -Path $LOG_FILE -Append -IncludeInvocationHeader
 
-"Not using encryption"
+if ([string]::IsNullOrWhiteSpace("${customer_master_key_id}")) {
+    "Not using encryption"
+} else {
+    "Using ecnryption key ${customer_master_key_id}"

```

Comment:
Typo on encryption

Reply:
i copy pasted this everywhere so there's going to be a lot of corrections.